From e8e85573de99c9a957f490efc7a07cd4860ac151 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Mon, 28 Mar 2016 23:59:01 +0300
Subject: [PATCH 318/538] dts: eip197: Add eip197 device support to CP-110
 device tree

Add entry in DTS file
Enabled in:
- armada-7040-rz-db-router.dts
- armada-7040-db-router.dts

Change-Id: I0826bdf432f45bbd329a4e497826467ff7d26bde
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28616
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yelena Krivosheev <yelena@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 Documentation/devicetree/bindings/crypto/mv_eip197.txt | 18 ++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-7040-db-router.dts  |  3 +++
 .../boot/dts/marvell/armada-7040-rz-db-router.dts      |  3 +++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi          |  7 +++++++
 4 files changed, 31 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/crypto/mv_eip197.txt

diff --git a/Documentation/devicetree/bindings/crypto/mv_eip197.txt b/Documentation/devicetree/bindings/crypto/mv_eip197.txt
new file mode 100644
index 0000000..d28f208
--- /dev/null
+++ b/Documentation/devicetree/bindings/crypto/mv_eip197.txt
@@ -0,0 +1,18 @@
+Marvell EIP197 Security Packet Engine with Classifiers
+
+Required properties:
+- compatible: should be one of the following string
+	      "marvell,eip197"
+- reg: base physical address of the engine and length of memory mapped region.
+- interrupts: interrupt number
+- clocks: One ore more clock gating clocks to be used to enable EIP197 unit in the SoC.
+
+Examples:
+
+eip197: eip197@800000 {
+	compatible = "marvell,eip197";
+	reg = <0x800000 0x200000>;
+	interrupts = <GIC_SPI 54 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&gateclk 26>;
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
index 87f42e3..3434746 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
@@ -106,6 +106,9 @@
 					status = "okay";
 				};
 			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 
 		pcie@0x640000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
index 6617e61..20ef7d6 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
@@ -105,6 +105,9 @@
 					status = "okay";
 				};
 			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 
 		pcie@0x640000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 8f8ec10..4736c85 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -221,3 +221,10 @@ ppv22@000000 {
 		status = "disabled";
 	};
 };
+eip197: eip197@800000 {
+	compatible = "marvell,eip197";
+	reg = <0x800000 0x200000>;
+	interrupts = <GIC_SPI 54 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&gateclk 26>;
+	status = "disabled";
+};
-- 
1.9.1

